### ISSUE TITLE
My exotic pet DISSAPEARED

### ISSUE THREAD

**Issue by Valdanif** [dalaranwow/dalaran-wow#5214]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
As a hunter, when i was changing my Primary Talents(Beast Master Spec) to Secondary Talents(Survival Spec) I lost my exotic pet, a pet that took me days to find and tame(pet: Loque'nahak). I looked into my stable, but it wasn't there. I understand that survival spec cannot use exotic pets, but there wasn't a reason to remove it from my stable.

**Expected behaviour**: Tell us what should happen instead
My exotic pet should've stored into my stable when changing Spec.

**Steps to reproduce the problem**:

1. 
2. 
3. 

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by TeridanDW** [comment]
Hello and thank you for the report @Valdanif,
Issue confirmed on PTR, as I was able to reproduce it.
Exotic pet should be stored in the next free stable slot (if there is one, if not just despawn it and on call pet give error to player that he can't use exotic pets) on spec switch, but currently it's not. And with a little bit of pet switch in none beast master spec the exotic pet is getting overwritten by a normal pet.

**Comment by Valdanif** [comment]
Thank you
